### PR TITLE
Fix XDocument test issues

### PR DESCRIPTION
**Pull request by sepidehMS**
There were some failing tests among System.Xml.XDocument tests. I split and fixed those tests.
There was also a threading issue in some of the tests. I added a lock and returned the stream to be read directly from the file.

### CODE REVIEW

File: src/Common/tests/System/Xml/XmlCoreTest/FilePathUtil.cs [hunk 1]
```diff
@@ -109,16 +109,20 @@ public static string ExpandVariables(string inputPath)
         private static MyDict<string, Stream> s_XmlFileInMemoryCache = null;
         private static MyDict<string, Stream> s_XmlFileInMemoryCacheBackup = null;
 
+        private static object s_XmlFileMemoryCacheLock = new object();
```
Comment (stephentoub) [inline src/Common/tests/System/Xml/XmlCoreTest/FilePathUtil.cs]: Nit: readonly

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs [hunk 2]
```diff
@@ -3400,7 +3401,9 @@ public void CData_4()
                 }
 
                 //[Variation(Id = 5, Desc = "WriteCData with ]]>", Priority = 1)]
-                public void CData_5()
+                [Fact]
+                [ActiveIssue(4054, default(PlatformID))]
```
Comment (stephentoub) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: What is the ```default(PlatformID)``` for?  Is the goal to disable this for all platforms?  If so, then just do ```[ActiveIssue(4054)]```... if instead the goal is to disable it for a specific platform, then you can use ```[ActiveIssue(4054, PlatformID.PlatformName)]```.

**Comment by stephentoub** [comment]
cc: @krwq, can you review?

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs (hunk 2 above)
Comment (sepidehMS) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: The goal was to annotate the test with ActiveIssue without disabling it. These are tests which are not failing, but are testing design decisions which need to be reevaluated.

File: src/Common/tests/System/Xml/XmlCoreTest/FilePathUtil.cs [hunk 3]
```diff
@@ -138,6 +142,18 @@ public static void cacheXml(string filename, string content)
...
+        public static Stream getStreamDirect(string filename)
+        {
+            foreach (var file in GetDataFiles())
+            {
+                if (file.Item1 == filename)
+                {
+                    return new MemoryStream(file.Item2);
+                }
+            }
+            throw new FileNotFoundException("File Not Found: " + filename);
```
Comment (krwq) [inline src/Common/tests/System/Xml/XmlCoreTest/FilePathUtil.cs]: nit: enter after `}`

File: src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs [hunk 4]
```diff
@@ -78,13 +78,39 @@ public XmlWriter CreateWriter(XDocument d)
         public XmlReader GetReader()
         {
             string file = Path.Combine("TestData", "XmlReader", "API", pGenericXml);
-            using (XmlReader r = XmlReader.Create(FilePathUtil.getStream(file), _rsx))
+            Stream s = FilePathUtil.getStream(file);
+
+            if (s == null)
+            {
+                TestLog.WriteLine("Could not find stream");
```
Comment (krwq) [inline src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs]: If this is fatal, throw an exception instead

File: src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs [hunk 5]
```diff
@@ -78,13 +78,39 @@ public XmlWriter CreateWriter(XDocument d)
...
 
+        public XmlReader GetPGenericXmlReader()
+        {
+            string file = Path.Combine("TestData", "XmlReader", "API", pGenericXml);
+            {
+                Stream s = FilePathUtil.getStreamDirect(file);
+
+                if (s == null)
+                {
+                    TestLog.WriteLine("getStream returned null");
```
Comment (krwq) [inline src/System.Xml.XDocument/tests/XDocument.Common/BridgeHelpers.cs]: same here

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs [hunk 6]
```diff
@@ -3603,7 +3608,7 @@ public void comment_6()
                         if (CompareReader(doc, "<Root><!--test - - --></Root>"))
                         {
                             throw new TestException(TestResult.Failed, "");
-                        }; //by design 329815
+                        }; //by design
```
Comment (krwq) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: nit: remove semicolon

**Comment by krwq** [comment]
Except for tiny issues LGTM :+1:

File: src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs (hunk 2 above)
Comment (sepidehMS) [inline src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs]: As we don't want to disable the test, I will remove the ActiveIssue attribute and as a marker, I will leave the issue id in a comment instead.

**Pull request merged**